test/widgets/search_ride_page_test.dart

```diff
@@ -0,0 +1,967 @@
+        await tester.tap(relaxRestrictionsButton);
+        await tester.pump();
+
+        expect(find.byKey(const Key('searchRideFilterDialog')), findsOneWidget);
+      });
+
+      testWidgets('Results at wrong times', (WidgetTester tester) async {
+        final DateTime searchTime = DateTime.now();
+        final DateTime rightTime = searchTime.add(const Duration(days: 2));
+        final DateTime farAwayTime = searchTime.add(const Duration(days: 4));
+
+        final List<Map<String, dynamic>> drives = [
+          DriveFactory()
+              .generateFake(startPosition: Position(0, 0), endPosition: Position(0, 0), startTime: farAwayTime)
+              .toJsonForApi(),
+          DriveFactory()
+              .generateFake(startPosition: Position(0, 0), endPosition: Position(0, 0), startTime: rightTime)
+              .toJsonForApi(),
+        ];
+
+        whenRequest(processor).thenReturnJson(drives);
+        whenRequest(processor, urlMatcher: matches(RegExp('/rest/v1/drives.*id=eq.')))
+            .thenReturnJson(DriveFactory().generateFake().toJsonForApi());
+
+        await pumpMaterial(tester, const SearchRidePage());
+
+        await enterStartAndDestination(tester, faker.address.city(), faker.address.city());
+        await enterDateAndTime(tester, searchTime);
+
+        expect(find.byType(RideCard, skipOffstage: false), findsNothing);
+
+        expect(find.byKey(const Key('searchRideWrongTime')), findsOneWidget);
+
+        final Finder wrongTimeButton = find.byKey(const Key('searchRideWrongTime')).hitTestable();
+        await tester.scrollUntilVisible(wrongTimeButton, 100, scrollable: find.byType(Scrollable).first);
+        await tester.tap(wrongTimeButton);
+        await tester.pump();
+
+        final SearchRidePageState pageState = tester.state(pageFinder);
+
+        expect(pageState.selectedDate, rightTime);
+        expect(find.byType(RideCard, skipOffstage: false), findsOneWidget);
+      });
+    });
+
+    group('Filter', () {
+      testWidgets('IndicatorRow', (WidgetTester tester) async {
+        await pumpMaterial(tester, const SearchRidePage());
+
+        final Finder indicatorRow = find.byKey(const Key('searchRideFilterButton'));
+
+        expect(indicatorRow, findsOneWidget);
+
+        await enterFilter(tester, rating: 3);
+        expect(find.descendant(of: indicatorRow, matching: find.byIcon(Icons.star)), findsOneWidget);
+
+        await enterFilter(tester, rating: 3, comfortRating: 3, hospitalityRating: 3);
+        expect(find.descendant(of: indicatorRow, matching: find.byIcon(Icons.star)), findsNWidgets(3));
+        expect(find.descendant(of: indicatorRow, matching: find.byIcon(Icons.chair)), findsOneWidget);
+        expect(find.descendant(of: indicatorRow, matching: find.byIcon(Icons.favorite)), findsOneWidget);
+        for (int i = 0; i < 2; i++) {
+          expect(find.descendant(of: indicatorRow, matching: find.byKey(Key('ratingSizedBox$i'))), findsOneWidget);
+        }
```

Comment:
lol this is longer than the two liens would have been

Reply:
Yes but I think it's okay. I think it's in line with some other test I copied it from